### PR TITLE
Refactor/mq doc and error handeling of theta

### PR DESCRIPTION
**Pull request by Javierverbel**
### Description
- Improve doc and error handling of the base problem and algorithm class

### Review process
- Verify that the value error handling is improved

### Pre-approval checklist
- [ ] The code builds clean without any errors or warnings
- [ ] I've added/updated tests
- [ ] I've added/updated documentation if needed

### CODE REVIEW

File: cryptographic_estimators/MQEstimator/mq_algorithm.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def __init__(self, problem: MQProblem, **kwargs):
         if q is not None and not is_prime_power(q):
             raise ValueError("q must be a prime power")
 
-        if w is not None and not 2 <= w <= 3:
+        if  w < 2 or 3 < w:
```
Comment (FloydZ) [inline cryptographic_estimators/MQEstimator/mq_algorithm.py:59]: the `or` should be an `and`, right?

File: cryptographic_estimators/MQEstimator/mq_problem.py [hunk 2]
```diff
@@ -36,12 +36,26 @@ class MQProblem(BaseProblem):
...
+    - ``theta`` -- exponent of the conversion factor (default: 2)
     - ``nsolutions`` --  number of solutions in logarithmic scale (default: max(expected_number_solutions, 0))
-    - ``memory_bound`` -- maximum allowed memory to use for solving the problem
+    - ``memory_bound`` -- maximum allowed memory to use for solving the problem (default: inf)
+
+    NOTE:
+
+    - If ``0 <= theta <= 2``, every multiplication in GF(q) is counted as `log2(q) ^ theta` binary operation.
+    - If ``theta = None``, every multiplication in GF(q) is counted as `2 * log2(q) ^ 2 + log2(q)` binary operation.
+
```
Comment (FloydZ) [inline cryptographic_estimators/MQEstimator/mq_problem.py:47]: shouldn't this comment added to the field `theta`, just above?

File: cryptographic_estimators/MQEstimator/mq_algorithm.py (hunk 1 above)
Comment (FloydZ) [reply cryptographic_estimators/MQEstimator/mq_algorithm.py:59]: and 3 and w should be swapped?

File: cryptographic_estimators/MQEstimator/mq_problem.py (hunk 2 above)
Comment (Javierverbel) [reply cryptographic_estimators/MQEstimator/mq_problem.py:47]: I think it is too long to be added there. What do you think?

File: cryptographic_estimators/MQEstimator/mq_algorithm.py (hunk 1 above)
Comment (FloydZ) [reply cryptographic_estimators/MQEstimator/mq_algorithm.py:59]: forget what I wrote: I overlooked the `not` in the previous line. Sorry

Comment (Javierverbel) [reply cryptographic_estimators/MQEstimator/mq_algorithm.py:59]: It is correct with an or. 
not 2 <= w <= 3 equals   w < 2 or 3 < w

File: cryptographic_estimators/MQEstimator/mq_problem.py (hunk 2 above)
Comment (FloydZ) [reply cryptographic_estimators/MQEstimator/mq_problem.py:47]: I mean:  it should be possible to write something like:
```
    - ``theta`` -- exponent of the conversion factor (default: 2)
         - If ``0 <= theta <= 2``, every multiplication in GF(q) is counted as `log2(q) ^ theta` binary operation.
         - If ``theta = None``, every multiplication in GF(q) is counted as `2 * log2(q) ^ 2 + log2(q)` binary operation.
    - ``nsolutions`` --  number of solutions in logarithmic scale (default: max(expected_number_solutions, 0))
```

Comment (Javierverbel) [reply cryptographic_estimators/MQEstimator/mq_problem.py:47]: Agree

**Review by FloydZ — APPROVED** [review]

**Pull request merged**